extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/DatasourceOperations.java

```diff
@@ -90,41 +91,59 @@ public void executeScript(String scriptFilePath) throws SQLException {
   }
 
   /**
-   * Executes SELECT Query
+   * Executes SELECT Query and returns the results after applying a transformer
    *
    * @param query : Query to executed
-   * @param entityClass : Class of the entity being selected
-   * @param transformer : Transformation of entity class to Result class
-   * @param entityFilter : Filter to applied on the Result class
-   * @param limit : Limit to to enforced.
-   * @return List of Result class objects
+   * @param converterInstance : An entity of the type being selected, used to convert to
+   *     PolarisBaseEntity
+   * @return The list of results yielded by the query
    * @param <T> : Entity class
    * @param <R> : Result class
    * @throws SQLException : Exception during the query execution.
    */
   public <T, R> List<R> executeSelect(
       @Nonnull String query,
-      @Nonnull Class<T> entityClass,
-      @Nonnull Function<T, R> transformer,
-      Predicate<R> entityFilter,
-      int limit)
+      @Nonnull Converter<T> converterInstance,
+      @Nonnull Function<T, R> transformer)
```

Comment:
It seems that we could construct a "result entity" directly from the `ResultSet`. There is no transformer is required, and converter will need to take one more type mark like `Converter<T, R>`.  Not a blocker though. We could make that change later.

Reply:
That's a good idea, I think we'd have to refactor `Converter` a little bit so that we could just call `converterInstance.convert` here or something. Might leave it for now since the goal here is just to pull the logic out of `executeSelectOverStream`